### ISSUE TITLE
thinking about validating the input/output buffer

### ISSUE THREAD

**Issue by psteinb** [mpicbg-scicomp/gearshifft#66]
in order to make gearshifft more versatile, we could think about comparing the input and the output buffer after a `iFFT(FFT(input)` roundtrip. This way gearshifft can not only be used for benchmarking but also for hardware checks on the memory subsystem. This is can be very useful.

**Comment by tdd11235813** [comment]
this is already done in [benchmark_executor.hpp](https://github.com/mpicbg-scicomp/gearshifft/blob/master/inc/core/benchmark_executor.hpp#L55) and [benchmark_data.hpp](https://github.com/mpicbg-scicomp/gearshifft/blob/master/inc/core/benchmark_data.hpp#L62).
One could argument, that an error of a single transform might be not visible if it becomes "corrected" on the inverse transform. For real hardware checks a second evaluation path executed on a different memory system must be considered, i.e. result of forward and inverse transform on device A must be compared with results gathered on device B. 

Since your suggestion is already implemented in gearshifft, we could rephrase it to a feature request where gearshifft should provide an interface to get the results of each transform to enable comparison.

**Comment by psteinb** [comment]
ok, that sounds reasonable. for a first try, I would however not open 
the API more. I think an automated way to cross check the transform 
results would be enough for now after all repetitions are done. the 
error bound should be configurable though (through a compile flag IMHO 
is enough)

**Comment by tdd11235813** [comment]
ah I knew I forgot a code line: the error bound and the number of runs can be changed in [application.hpp](https://github.com/mpicbg-scicomp/gearshifft/blob/master/inc/core/application.hpp#L23). At the moment these are template values, so required at compile-time.

I will come back to your feature request at a later time. I also noted to extend the README on how error checking works in gearshifft. The idea to use a gearshifft wiki becomes stronger ...
